### PR TITLE
Disable conflicting search fields when fields are changed

### PR DESCRIPTION
**Pull request by maurizi**
This prevents searches that don't make much sense, such as searching for
a specific species and a tree missing a species at the same time.

Connects to #2420

screenshot from 2016-03-31 22-49-44
screenshot from 2016-03-31 22-50-11
screenshot from 2016-03-31 23-00-14

### CODE REVIEW

**Comment by RickMohr** [comment]
Hey, now we can remove the (lame) message "Checking a field's box below overrides any other filter for that field"!

**Comment by RickMohr** [comment]
For fields not in the "missing" category I think a message like this would be better:

`Disabled while a "Missing" search for this field is active.`

File: opentreemap/treemap/js/src/searchBar.js [hunk 1]
```diff
@@ -224,18 +238,63 @@ function updateDisabledFieldGroups(search) {
...
+    // Then disable all fields which are not filled in but which have the same
+    // key as another filled in field
+    _.each(search.filter, function(predicate, field) {
+        var searchTypes = _.keys(predicate),
+            minOrMax = _.contains(searchTypes, 'MIN') || _.contains(searchTypes, 'MAX');
+        $(dom.searchFields).filter('[name="' + field + '"]').each(function(i, elem) {
+            var $elem = $(elem),
+                searchType = $elem.attr('data-search-type');
+
+            // Min/Max fields don't affect other min/Max fields
```
Comment (RickMohr) [inline opentreemap/treemap/js/src/searchBar.js]: I didn't get this on first reading. Maybe something like "e.g. when user fills in the min field, don't disable the corresponding max field."

**Comment by RickMohr** [comment]
And since there can only be one search for a field I think the other should say:

`Disabled while a search for this field is active`

File: opentreemap/treemap/js/src/searchBar.js [hunk 2]
```diff
@@ -224,18 +238,63 @@ function updateDisabledFieldGroups(search) {
...
+            if (minOrMax && _.contains(['MIN', 'MAX'], searchType)) {
+                return;
+            }
+
+            if (($elem.is(':checkbox') && !$elem.is(':checked')) || $elem.val().length === 0) {
```
Comment (RickMohr) [inline opentreemap/treemap/js/src/searchBar.js]: If I check the "Missing" box for a multiselect field I get a JS error `"Cannot read property 'length' of null"`

**Comment by RickMohr** [comment]
The "disabled" message is indented (and thus wraps) when the dropdown has a scrollbar:
image

**Comment by RickMohr** [comment]
"Disabled" message is indented (and thus wraps) on text inputs:
image

**Comment by RickMohr** [comment]
Otherwise works great.

**Comment by maurizi** [comment]
@RickMohr all fixed

**Comment by RickMohr** [comment]
To me, this is confusing to parse:
`Disabled while a missing search for this field is active`
(it sounds like the search is missing) which is why I'd prefer:
`Disabled while a "Missing" search for this field is active`

**Comment by RickMohr** [comment]
+1 with consideration of quote fix. Yay!

**Pull request merged**